File: frontend/src/types/market.ts
import { Address } from './address';

export type MarketType = 'physical' | 'online' | 'hybrid';
export type MarketStatus = 'active' | 'inactive';

export interface MarketSettings {
  orderPrefix?: string;   // Prefix for order numbers
  taxRate?: number;       // Default tax rate
  shippingZones?: string[];
}

export interface Market {
  id: string;
  tenantId: string;
  name: string;
  code: string;           // Unique within tenant
  type: MarketType;
  status: MarketStatus;
  currency: string;
  timezone: string;
  address?: Address;
  settings?: MarketSettings;
  apiKeyCount: number;    // Count of active API keys
  createdAt: Date;
  updatedAt: Date;
  createdBy?: string;
  updatedBy?: string;
}

export interface CreateMarketInput {
  tenantId: string;
  name: string;
  code: string;
  type: MarketType;
  currency: string;
  timezone: string;
  address?: Address;
  settings?: MarketSettings;
}

export interface UpdateMarketInput {
  name?: string;
  code?: string;
  type?: MarketType;
  status?: MarketStatus;
  currency?: string;
  timezone?: string;
  address?: Address;
  settings?: MarketSettings;
}

export interface MarketFilters {
  tenantId?: string;
  search?: string;
  type?: MarketType;
  status?: MarketStatus;
  page?: number;
  limit?: number;
}

File: api/EComm.Data/Entities/Market.cs
using EComm.Data.ValueObjects.Common;
using EComm.Data.ValueObjects.Tenant;

namespace EComm.Data.Entities;

public class Market
{
    public string Id { get; set; } = string.Empty;
    public string TenantId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public string Type { get; set; } = "physical";  // physical, online, hybrid
    public string Status { get; set; } = "active";
    public string Currency { get; set; } = "USD";
    public string Timezone { get; set; } = string.Empty;
    public Address? Address { get; set; }
    public MarketSettings? Settings { get; set; }
    public int ApiKeyCount { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
